src/main/java/marquez/db/DatasourceDao.java

```diff
@@ -28,13 +28,16 @@
 @RegisterRowMapper(DatasourceRowMapper.class)
 public interface DatasourceDao {
   @SqlUpdate(
-      "INSERT INTO datasources (guid, name, connection_url) "
-          + "VALUES (:uuid, :name, :connectionUrl)")
-  void insert(@BindBean DatasourceRow dataSourceRow);
+      "INSERT INTO datasources (guid, urn, name, connection_url) "
+          + "VALUES (:uuid, :urn, :name, :connectionUrl)")
```

Comment:
To avoid the _insert-then-get_ pattern in `DatasourceService`, use:

```java
@SqlQuery("INSERT INTO datasources (guid, urn, name, connection_url) "
	          + "VALUES (:uuid, :urn, :name, :connectionUrl"
                  + "RETURNING *")
Optional<DatasourceRow> insertAndGet(@BindBean DatasourceRow datasourceRow)
```

I'd double check my query.

see: https://www.postgresql.org/docs/9.6/dml-returning.html

Reply:
This is a really cool idea that would simplify our code, but inserts as part of the `SQLQuery` annotation seem to not accept this type of output.
When I tried adding the `RETURNING *` clause, I received this from my dao test: 
```
org.jdbi.v3.sqlobject.UnableToCreateSqlObjectException: DatasourceDao.insert method is annotated with @SqlUpdate so should return void, boolean, or Number but is returning: java.util.Optional<marquez.db.models.DatasourceRow>
```

More info: https://github.com/jdbi/jdbi/blob/master/sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/SqlUpdateHandler.java#L35-L69